src/main/java/com/microsoft/sqlserver/jdbc/spark/DataFrameBulkRecord.java

```diff
@@ -37,7 +37,9 @@ public DataFrameBulkRecord(Iterator<Row> iterator, ColumnMetadata[] dfColumnMeta
              // Columns may be reordered between SQLTable and DataFrame. dfFieldIndex maps to the
              // corresponding column in rowData and thus use dfFieldIndex to get the column.
              int dfFieldIndex = dfColumnMetadata[idx].getDfColIndex();
-             rowData[idx] = row.get(dfFieldIndex);
+             if (dfFieldIndex != -1) {
```

Comment:
important that special handling is clear from code. Here the reader will be confused by why will a col have a dfFieldIndex = -1

Reply:
Now using isAutoIncrement instead of dfFieldIndex = -1